src/solver/pad_constant/pad_constant.cpp

```diff
@@ -0,0 +1,211 @@
+            for(auto i = 0; i < params.padding_size; i++)
+                padding.val[i] = params.padding[i];
+
+            size_t output_size = params.yDesc->GetElementSize();
+
+            kernel(params.x,
+                   params.y,
+                   input_tv,
+                   output_tv,
+                   padding,
+                   output_size,
+                   params.padding_value);
+        };
+    };
+
+    return result;
+}
+} // namespace pad_constant_fwd
+
+namespace pad_constant_bwd {
+
+bool PadConstantBwd::IsApplicable(const ExecutionContext& /*context*/,
+                                  const miopen::pad_constant_bwd::ProblemDescription& problem) const
+{
+    if(!problem.IsSameType())
+        return false;
+    if(!problem.IsSameShape())
+        return false;
+    if(!problem.IsPaddingValid())
+        return false;
+    if(!problem.IsImprovementOverROCm())
+        return false;
+
+    return true;
+}
+
+ConvSolution
+PadConstantBwd::GetSolution(const ExecutionContext& /*context*/,
+                            const miopen::pad_constant_bwd::ProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto input_dtype  = miopen::GetDataType(problem.GetXDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetYDesc().GetType());
+
+    size_t input_size = problem.GetXDesc().GetElementSize();
+
+    size_t xlocalsize = 256;
+    size_t xgridsize  = AlignUpUL(input_size, xlocalsize);
+    size_t ylocalsize = 1;
+    size_t ygridsize  = 1;
+
+    auto kernel = KernelInfo{};
+
+    kernel.kernel_file = "MIOpenPadConstant.cpp";
+    if(problem.IsContiguous())
+        kernel.kernel_name = "PadConstantBwdContiguous";
+    else
+        kernel.kernel_name = "PadConstantBwd";
+
+    const auto build_params = KernelBuildParameters{
+        {"INPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+        {"OUTPUT_TYPE", output_dtype == "bfloat16" ? "ushort" : output_dtype},
+        {"MIOPEN_USE_FP64", static_cast<int>(output_dtype == "double")},
+        {"MIOPEN_USE_FP32", static_cast<int>(output_dtype == "float")},
+        {"MIOPEN_USE_FP16", static_cast<int>(output_dtype == "half")},
+        {"MIOPEN_USE_BFP16", static_cast<int>(output_dtype == "bfloat16")},
+    };
+
+    kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+    kernel.l_wk.push_back(xlocalsize);
+    kernel.l_wk.push_back(ylocalsize);
+
+    kernel.g_wk.push_back(xgridsize);
+    kernel.g_wk.push_back(ygridsize);
+
+    result.construction_params.push_back(kernel);
+
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& invoke_params) {
+            decltype(auto) kernel = handle_.Run(kernels.front());
+            decltype(auto) params = invoke_params.CastTo<miopen::pad_constant_bwd::InvokeParams>();
+
+            tensor_view_5d_t input_tv  = get_inner_expanded_tv(*params.dxDesc);
```

Comment:
It seems strange that dx is input. Please change value name.

Reply:
I've changed it as requested